src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -191,16 +191,44 @@ public void Initialize(string fileName)
             {
                 // failed to find the VS Service
             }
-            string natvisPath = null;
-            string directory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
-            if (!string.IsNullOrEmpty(directory) && !string.IsNullOrEmpty(fileName))
+
+            if (string.IsNullOrEmpty(Path.GetDirectoryName(fileName)))
             {
-                natvisPath = Path.Combine(directory, fileName);
-            }
+                string globalVisualizersDirectory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
+                string globalNatVisPath = null;
+                if (!string.IsNullOrEmpty(globalVisualizersDirectory) && !string.IsNullOrEmpty(fileName))
+                {
+                    globalNatVisPath = Path.Combine(globalVisualizersDirectory, fileName);
+                }
 
-            if (!string.IsNullOrEmpty(natvisPath))
+                // For local launch, try and load natvis next to the target exe if it exists.
+                if (_process.LaunchOptions is LocalLaunchOptions)
+                {
+                    string natvisPath = Path.Combine(Path.GetDirectoryName((_process.LaunchOptions as LocalLaunchOptions).ExePath), fileName);
+
+                    if (File.Exists(natvisPath))
+                    {
+                        LoadFile(natvisPath);
+                        return;
+                    }
+                    else if (!File.Exists(globalNatVisPath))
```

Comment:
Unfortunately, the scenario has been a bit bastardized by the existing launch options. In the vs case, we expect the natvis file to come from solution or from the global path. For vscode, we expect it to come from either a fullpath, a workspace relative path (if possible in the local case) or the global path. I'd love to revisit this, but the cost seems too high now (changes across too many products). 

I agree the code isn't all that readable, but it seems to be the best I can do to capture the behavior I described above. Perhaps I can make it a bit more readable.